src/text-editor.coffee

```diff
@@ -2232,6 +2233,17 @@ class TextEditor extends Model
   # the function with merging suppressed, then merges intersecting selections
   # afterward.
   mergeIntersectingSelections: (args...) ->
+    @mergeSelections args..., (previousSelection, currentSelection) ->
+      exclusive = not currentSelection.isEmpty() and not previousSelection.isEmpty()
+
+      previousSelection.intersectsWith(currentSelection, exclusive)
+
+  mergeIntersectingSelectionsByRow: (args...) ->
```

Comment:
The code is great here, but I'm sticking on the naming. The selections we'll merge aren't actually always intersecting, right? They're intersecting-by-row, which feels like a mouthful. What about a name like `mergeSelectionsOnSameRows`?